### PR TITLE
Unify warnings for using node v5.2 and <v0.12 through debugger ui

### PR DESCRIPTION
**Pull request by matthewloring**
The agent is currently crashing on node 0.10 because
EventEmitter.prototype is undefined which breaks inheritance for the
debuglet. This patch causes the agent to shutdown immediately if running
on an unsupported version of node. It also addresses #7 since the
version check runs first and it is now easier to reason about graceful
exit in this case.

### CODE REVIEW

**Comment by matthewloring** [comment]
The other option I see here is to add special logic in the debuglet to handle the case where it is running in 0.10. I would rather shut down the agent immediately in this case as it communicates the issue to the user more immediately than waiting until a breakpoint set is attempted. @ofrobots WDYT?

File: index.js [hunk 1]
```diff
@@ -39,11 +38,13 @@ if (process.env.hasOwnProperty('GCLOUD_DEBUG_REPO_APP_PATH')) {
 
 var log = logger.create(config.logLevel, '@google/cloud-debug');
 
-if (semver.satisfies(process.version, '5.2')) {
-  log.error('Debug is not supported on Node v5.2');
-  return;
+if (semver.satisfies(process.version, '5.2 || <0.12')) {
+  log.error('Debug is not supported on Node v5.2 or <v0.12');
+  process.exit(1);
```
Comment (ofrobots) [inline index.js]: I am not so sure about the `process.exit`. It is quite possible for users to have dev environments different from the version of Node they would get on deployment. We are converting a lower severity problem ('my debug agent isn't starting') to a very high severity problem ('my app doesn't deploy').

**Comment by ofrobots** [comment]
I would be fine with this change if only we would use `return` instead of `process.exit` on unsupported versions.

File: index.js (hunk 1 above)
Comment (matthewloring) [inline index.js]: That makes sense. I've downgraded back to a `return`.

**Comment by matthewloring** [comment]
This has been updated to report all version failures at the time the first breakpoint is set as per offline discussion. Additionally, we no longer attempt to test on 0.10 as the package is not installable without npm support for namespaces. @ofrobots PTAL.

File: lib/debuglet.js [hunk 2]
```diff
@@ -30,9 +31,9 @@ var StatusMessage = require('./apiclasses.js').StatusMessage;
 
 var assert = require('assert');
 
-var NODE_VERSION_MESSAGE = 'V8 Debug API not available – make sure you are ' +
-    'running Node.js >=0.12. Cloud Debugger agent will not be able to take ' +
-    'snapshots.';
+var NODE_VERSION_MESSAGE = 'An unsupported version of Node.js is running. ' +
+  'Cloud Debugger is only supported with Node.js >=0.12 with the exception of ' +
+  ' 5.2.';
```
Comment (ofrobots) [inline lib/debuglet.js]: How about rewording this to 'Node.js version not supported. Node.js 5.2.0 and versions older than 0.12 are not supported.'

File: test/e2e/test-error-message.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright 2015 Google Inc. All Rights Reserved.
```
Comment (ofrobots) [inline test/e2e/test-error-message.js]: 2016

**Comment by ofrobots** [comment]
test/.DS_store should not be committed to the repository. Does our .gitignore not prevent this from happening already?

Secondly, our README.md needs to mention that 5.2.0 is not suppoted.

LGTM w/ these comments.

File: test/e2e/test-error-message.js (hunk 3 above)
Comment (matthewloring) [inline test/e2e/test-error-message.js]: Done.

File: lib/debuglet.js (hunk 2 above)
Comment (matthewloring) [inline lib/debuglet.js]: Sounds good.

**Comment by matthewloring** [comment]
The .gitignore has been updated and the other comments have been addressed.
